### DISCUSSION TITLE
Using non-stacked packages that requires context

### DISCUSSION THREAD

**Discussion by dvagala** [FilledStacks/stacked#924] [General]
Hey! I have a question :)

Many non-stacked packages requires context to trigger something on the UI. For example `tutorial_coach_mark` that show overlay tutorial on screen. Naturally I want to use these packages in the `viewModel` or `service` where I don't have access to context. What do you think is a preferred and clean way to tackle this problem? 

Pass the context as a parameter to the nested `viewModel` and `service` methods?
Or store the context in some service and access it there?
Or any better solution?

**Comment by FilledStacks** [comment]
@dvagala this is a UI only thing so I would build it all in the view file. Some parts you can rely on from the ViewModel like the strings or anything else, but I'd suggest you build this in a stateful widget. If that gets too messy simply make a mixin and mix that into your stateful widget to hide some of the implementation details and move it away from the UI rendering. 

In terms of passing build context to ViewModel or Service. That should never be done, it's the complete opposite of what Stacked wants to achieve, which is separate state from the Flutter UI rendering framework.
